### ISSUE TITLE
Regression in pulumi command makes command.output secret failing examples

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-command#256]
### What happened?

Copying from: https://github.com/pulumi/examples/issues/1491

TestAccAwsPyEc2Provisioners test is failing in CI on master. 

I've looked into this and here's the problem.  There's two versions of this program, one in TS and one in Python.

With recent changes to pulumi-command, if you are on 0.8.2, the following program:

```typescript
// Execute a basic command on our server.
const catConfig = new command.remote.Command("cat-config", {
    triggers: [changeToken],
    connection,
    create: "cat myapp.conf",
}, { dependsOn: cpConfig });

export const publicIp = server.publicIp;
export const publicHostName = server.publicDns;
export const catConfigStdout = catConfig.stdout;
```

now marks catConfig.Stdout as a secret output whereas it didn't before.

Now, Node test pins Pulumi-command 0.0.3 or some such, while Python test floats to the latest reference. Therefore Node test passes but Python test fails. 

The specific failure is a panic when reading the secret value in ProgramTest Go code. There's an assert that tries to convert catConfigStdout to a string, but it is now a map in Go land.  AFAIK Go ProgramTest cannot yet peek inside secret values returned from the stack.
 

### Expected Behavior

Passing test

### Steps to reproduce

N/A

### Output of `pulumi about`

N/A

### Additional context

We had some discussion and dug up this change https://github.com/pulumi/pulumi-go-provider/pull/105/files#diff-4bd4b3f761d7688f63a5a8e766057562d08e3df2f51cfa88e604cb4e353fd042R318 as possible root cause.

**Need to decide** if this semantic is reasonable for Command and we want to keep it or revert back. If keeping this semantic, we need to adjust the test here to not fail on receiving a secret.

I think the examples issue is fixed by pinning older Command version which is unsustainable so if we decide to keep this behavior we should update the test to unpin.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
It sounds like we need to decide whether this is desired behaviour or not. I'll mark this issue as P1 as a regression for now - until we close it one way or the other.

@iwahbe Can you summarize the change that caused the issue and the reasoning behind making `output` a secret?

**Comment by t0yv0** [comment]
I think connection is secret and by the new go-provider behavior, any secrets in the input make all outputs secret. This doesn't make sense in this case IMO. Connection is secret does not imply "myapp.conf" contents are secret.
